api/src/migrations/convert-translation-messages.js

```diff
@@ -0,0 +1,82 @@
+const _ = require('underscore'),
+  { promisify } = require('util'),
+  db = require('../db-nano'),
+  properties = require('properties');
+
+module.exports = {
+  name: 'convert-translation-messages',
+  created: new Date(2018, 11, 8),
+  run: promisify((callback) => {
+	  db.request({
+      db: db.settings.db,
+      method: 'GET',
+      path: '_design/medic',
+    },
+    (err, result) => {
+      if (err) {
+        return callback(err);
+      }
+
+      let translationAttachmentKeys = Object.keys(result._attachments).filter(k => k.includes('translations'));
+      translationAttachmentKeys.forEach((translationAttachmentKey) => {
+        let translationMessageKey = translationAttachmentKey.match(/translations\/(.+).properties/)[1];
+        let translationMessageKeys = [translationMessageKey, translationMessageKey + '-backup'];
+        translationMessageKeys.forEach((translationMessageKey) => {
+          db.request({
+            db: db.settings.db,
+            method: 'GET',
+            path: translationMessageKey,
+          },
+          (err, translationMessageValues) => {
+            if (_.has(translationMessageValues, 'values')) {
+              db.request({
+                db: db.settings.db,
+                method: 'GET',
+                path: '_design/medic/' + translationAttachmentKey,
+              },
+              (err, translationAttachmentValue) => {
+                if (err) {
+                  return callback(err);
+                }
+
+                properties.parse(translationAttachmentValue.toString('utf8'), (err, translationAttachmentValues) => {
+                  if (err) {
+                    return callback(err);
+                  }
+
+                  let defaultTranslations = {};
+                  let customTranslations = {};
+                  Object.keys(translationMessageValues.values).forEach((translationMessageKey) => {
+                    if (_.has(translationAttachmentValues, translationMessageKey)) {
+                      defaultTranslations[translationMessageKey] = translationMessageValues.values[translationMessageKey];
+                    } else {
+                      customTranslations[translationMessageKey] = translationMessageValues.values[translationMessageKey];
+                    }
+                  });
+
+                  delete translationMessageValues.values;
+                  translationMessageValues.custom = customTranslations;
+                  translationMessageValues.default = defaultTranslations;
```

Comment:
`default` is a keyword in javascript so it's probably not a good property name. Can we come up with another name for it?

Reply:
Switched to `generic`